datafusion/physical-plan/src/lib.rs

```diff
@@ -235,7 +235,28 @@ pub trait ExecutionPlan: Debug + DisplayAs + Send + Sync {
         Ok(None)
     }
 
-    /// Begin execution of `partition`, returning a stream of [`RecordBatch`]es.
+    /// Start plan execution for `partition`, returning a [`SendableRecordBatchStream`] of [`RecordBatch`]es.
+    ///
+    ///
+    /// Often its needed to call [`.await`](https://doc.rust-lang.org/std/keyword.await.html) just after the stream async combinators.
+    /// To call it within the sync method please refer the snippet below. The snippet can also be used to help output/debug the node execution on batch level
+    ///
+    /// ```no_run
+    /// use datafusion_common::DataFusionError;
+    ///
+    /// fn print_plan_exec() -> Result<(), DataFusionError> {
+    ///     let stream = node.execute(0, task_ctx)?;
+    ///     futures::stream::once(async move {
```

Comment:
This stream gets immediately dropped and therefore will never run?

Reply:
okay, I'll include another use case with amending stream and returning it back to the user. Like in https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-plan/src/sorts/sort.rs#L895

@tustvold I believe you mean exactly that?

Reply:
I mean as written this example doesn't do anything and is therefore just misleading.

> Like in https://github.com/apache/arrow-datafusion/blob/main/datafusion/physical-plan/src/sorts/sort.rs#L895

Yes, that is I think a good example of the pattern